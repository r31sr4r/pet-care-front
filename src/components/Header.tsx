import { AccountCircle } from '@mui/icons-material';
import MenuIcon from '@mui/icons-material/Menu';
import {
	AppBar,
	Box,
	Button,
	IconButton,
	Menu,
	MenuItem,
	Toolbar,
	Typography,
} from '@mui/material';
import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import logo_white from '../assets/images/logo-white.png';
import { UserData } from '../utils/security/UserData';
import Brightness4Icon from '@mui/icons-material/Brightness4';
import Brightness7Icon from '@mui/icons-material/Brightness7';

export function Header({
	toggle,
	theme,
}: {
	toggle: () => void;
	theme: string;
}) {
	let navigate = useNavigate();
	const token = localStorage.getItem('token');
	const [userLogged, setUserLogged] = React.useState<boolean>(token !== null);
	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

	const handleLogout = () => {
		setUserLogged(false);
		localStorage.removeItem('token');
		navigate('/signin');
	};

	const handleLogin = () => {
		navigate('/signin');
	};

	useEffect(() => {
		setUserLogged(token !== null);
	}, [token]);

	const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
		setAnchorEl(event.currentTarget);
	};

	const handleClose = (ev: any) => {
		setAnchorEl(null);
		if (ev.nativeEvent.target.outerText === 'Sair') {
			handleLogout();
		} else if (ev.nativeEvent.target.outerText === 'Alterar Senha') {
			const id = UserData()?.user.id;
			navigate(`/user/change-password`);
		}
	};

	return (
		<Box sx={{ flexGrow: 1 }}>
			<AppBar position="static">
				<Toolbar>
					<IconButton
						size="large"
						edge="start"
						color="inherit"
						aria-label="menu"
						sx={{ mr: 2 }}
					>
						<MenuIcon />
					</IconButton>
					<div>
						<img src={logo_white} alt="pet-care-club" />
					</div>
					<Typography
						variant="h6"
						component="div"
						sx={{ ml: 1, flexGrow: 1 }}
					>
						Pet Care Club
					</Typography>

					<IconButton
						sx={{ ml: 1 }}
						onClick={toggle}
						color="inherit"
					>
						{theme === 'dark' ? (
							<Brightness7Icon />
						) : (
							<Brightness4Icon />
						)}
					</IconButton>

					{!userLogged ? (
						<Button color="inherit" onClick={handleLogin}>
							Login
						</Button>
					) : (
						<div>
							<IconButton
								size="large"
								aria-label="account of current user"
								aria-controls="menu-appbar"
								aria-haspopup="true"
								onClick={handleMenu}
								color="inherit"
							>
								<AccountCircle />
							</IconButton>
							<Menu
								id="menu-appbar"
								anchorEl={anchorEl}
								anchorOrigin={{
									vertical: 'top',
									horizontal: 'right',
								}}
								keepMounted
								transformOrigin={{
									vertical: 'top',
									horizontal: 'right',
								}}
								open={Boolean(anchorEl)}
								onClose={handleClose}
							>
								<MenuItem onClick={handleClose}>
									Alterar Senha
								</MenuItem>
								<MenuItem onClick={handleClose}>
									Minha Conta
								</MenuItem>
								<MenuItem onClick={handleClose}>Sair</MenuItem>
							</Menu>
						</div>
					)}
				</Toolbar>
			</AppBar>
		</Box>
	);
}
