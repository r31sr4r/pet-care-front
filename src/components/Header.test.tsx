import { render } from '@testing-library/react';
import { Header } from './Header';
import { BrowserRouter } from 'react-router-dom';

describe('Header', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<Header             
            toggle={() => {}}
            theme="dark"
            handleDwawerToggle={() => {}}
             />, {
			wrapper: BrowserRouter,
		});
        expect(asFragment()).toMatchSnapshot();
    });
});