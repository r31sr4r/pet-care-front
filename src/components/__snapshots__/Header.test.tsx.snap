// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-w9hjgi-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div>
          <img
            alt="pet-care-club"
            src="logo-white.png"
          />
        </div>
        <div
          class="MuiTypography-root MuiTypography-h6 css-1q06x6u-MuiTypography-root"
        >
          Pet Care Club
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-whjmaz-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="Brightness7Icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Login
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
